src/SACA.chpl

```diff
@@ -0,0 +1,230 @@
+module SACA{
+// In this module, different algorithms to construct suffix array are provided
+//Nov.15, 2020
+//Algorithm 1
+// The first algorithm divsufsort is the fastest C codes on suffix array
+require "../thirdparty/SA/libdivsufsort/include/config.h";
+require "../thirdparty/SA/libdivsufsort/include/divsufsort.h";
+require "../thirdparty/SA/libdivsufsort/include/divsufsort_private.h";
+require "../thirdparty/SA/libdivsufsort/include/lfs.h";
+
+require "../thirdparty/SA/libdivsufsort/lib/divsufsort.c";
+require "../thirdparty/SA/libdivsufsort/lib/sssort.c";
+require "../thirdparty/SA/libdivsufsort/lib/trsort.c";
+require "../thirdparty/SA/libdivsufsort/lib/utils.c";
+extern proc divsufsort(inputstr:[] uint(8),suffixarray:[] int(32),totallen:int(32));
+
+//Another possible SACA algorithm to utilize. 
+//require "../thirdparty/SA/SACA-K/saca-k.c";
+
+//extern proc SACA_K(inputstr:[] uint(8), suffixarray:[] uint, n:uint, K:uint,m:uint, level:int);
+//void SACA_K(unsigned char *s, unsigned int *SA,
+//  unsigned int n, unsigned int K,
+//  unsigned int m, int level) ;
+
+//Algorithm 2
+
+// The Chapel version of suffix array construction algorithm using skew algorithm
+// Rewrite the algorithm and codes in paper
+// "Simple Linear Work Suffix Array Construction" by Juha Karkkainen and Peter Sanders (2003)
+// Dec.7, 2020
+
+inline proc leq(a1 :int, a2:int, b1:int, b2:int) // lexicographic order
+{  return(a1 < b1 || a1 == b1 && a2 <= b2); 
```

Comment:
@zhihuidu, I noticed you have an external C implementation of suffix array construction (Algorithm 1) and a native implementation (Algorithm 2). Have you done performance testing to compare the native implementation with the C implementation? If the native implementation is fast, there's no need for us to force the `libdivsufsort` dependency on Arkouda users

Reply:
@pierce314159  yes, you are right! Currently the performance of libdivsufsort is better. However it has two problems. (1) it cannot use multiple locales. (2)  we have to make some change on the code to interate it into Arkouda. We hope to develop a new multilocal algorithm to handle very large strings in the next step. For small strings, the performance difference in several seconds or miliseconds is not important. The major reason we use libdivsufsort is that we want to check how to use external code in Arkouda.

Reply:
That makes sense! For more info on how to use external code, you can checkout @glitch's [Investigating Arrow wiki page](https://github.com/glitch/arkouda/wiki/Investigating-Arrow). This is how I was planning to import the library instead of having a copy in our codebase.

In general, I think we want to keep the number of dependencies low. And native chapel implementations are almost always preferred because they can better leverage HPC resources